### PR TITLE
Week-4-PR-1: Created class to return time ranges based on a collection of events and requests

### PR DESCRIPTION
**Pull request by gracieorozco**
I used multiple if statements, for loops, and collections to manage and organize the data. I have also added comments to help explain what my code does. When I run mvn test, it passes all the tests, but when I run mvn appengine:devserver, it does not always work. For some names, it works fine, but others it doesn't. I am planning to improve this before the due date.

### CODE REVIEW

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,95 @@
...
+        for (Event e : event_list) {
+            start_time_list.add(e.getWhen().start());
+            end_time_list.add(e.getWhen().end());
+        }
+        // If the starting time is not 0, then create the first time range from
+        // the morning to the first element of start_time_list. 
+        if (start_time_list.get(0) != 0) {
+            multiple_time_range_list.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, start_time_list.get(0), false));
+        }
+        // Loop to check to if the first element in end_time_list at is less
```
Comment (KaylaNguyen) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: to see if*
at i*

**Review by KaylaNguyen — COMMENTED** [review]
Good job Gracie! Did you run this against all tests?

Btw, you can exclude irrelevant changed files like pom.xml or SentimentScore.java by checking out HEAD branch first before creating a new branch and making week 4 changes in that new branch.

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,95 @@
...
+        // the morning to the first element of start_time_list. 
+        if (start_time_list.get(0) != 0) {
+            multiple_time_range_list.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, start_time_list.get(0), false));
+        }
+        // Loop to check to if the first element in end_time_list at is less 
+        // than the start_time_list at i+1. If the statement is true, then
+        // a time range is added ranging from the element of end_time_list at 
+        // i to the start_time i+1. Otherwise, the for loop skips it. This
+        // loop checks until the second to last element in end_time_list. 
+        for (int i = 0; i < end_time_list.size()-1; i++) {
```
Comment (KaylaNguyen) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: This works, but we also have some existing function that you might want to check out: https://github.com/KaylaNguyen/SPS/blob/master/walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/TimeRange.java#L82

**Comment by gracieorozco** [comment]
I wasn't sure what you meant by checking out the HEAD branch, but I thought that rebasing the branch would remove the pom.xml and SentimentalScore.java files since they look the same on the master branch. I rebased the branch on the recent changes I made in master. The first rebase was difficult for me to do in the command line. I used the online editor instead to help me out in the second rebase. Since I formatted the code with the sps-clang-format, other files were changed as well.

File: walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,99 @@
...
+      if (start_time_list.get(0) != 0) {
+        multiple_time_range_list.add(
+            TimeRange.fromStartEnd(TimeRange.START_OF_DAY, start_time_list.get(0), false));
+      }
+      // Loop to check to see if the first element in end_time_list at i is less
+      // than the start_time_list at i+1. If the statement is true, then
+      // a time range is added ranging from the element of end_time_list at
+      // i to the start_time i+1. Otherwise, the for loop skips it. This
+      // loop checks until the second to last element in end_time_list.
+      for (int i = 0; i < end_time_list.size() - 1; i++) {
```
Comment (KaylaNguyen) [inline walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:87]: This is fine for this iteration. I would recommend using built-in method of TimeRange class to find overlapping time range.

**Review by KaylaNguyen — APPROVED** [review]

**Pull request merged**